src/Dimensions/primitives.jl

```diff
@@ -616,6 +622,10 @@ end
         isnothing(msg) || _orderaction(msg, a, b)
         return false
     end
+    if valtypename && Base.typename(typeof(parent(a))) != Base.typename(typeof(parent(b)))
```

Comment:
We could introduce a function for this check and return false as fallback and return true for combinations that assume to be comparable

Reply:
Still a bit complicated to know what the function should do. 

We may need to just manually distinguish AbstractSampled from AbstractCategorical, and let `promote_dims` take care of the rest

Reply:
Maybe this function could return true for identical types, or for combinations where there is some specialized implementation of `promote_first` (which is I think just for both `AbstractSampled` or two `AbstractCategorical`)

Reply:
The reason it returns this NoLookup currently is because nothing else is implemented for `promote_dims`/`promote_first` (I think?), so it would make sense to kind of mirror that.

Reply:
Yeah. Its a bit odd to have the same thing in two places... I wonder if we can somehow make it share the same code.